Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/WatchManager.java

```diff
@@ -99,28 +105,43 @@
 
     private final DatabaseManager databaseManager;
     private final DatabasePinManager databasePinManager;
+    private final DatabaseSavedThreadManager databaseSavedThreadManager;
     private final ChanLoaderFactory chanLoaderFactory;
     private final WakeManager wakeManager;
     private final PageRequestManager pageRequestManager;
+    private final ThreadSaveManager threadSaveManager;
 
     private IntervalType currentInterval = IntervalType.NONE;
 
     private final List<Pin> pins;
+    private final List<SavedThread> savedThreads;
+    private boolean prevIncrementalThreadSavingEnabled = false;
+
     private Map<Pin, PinWatcher> pinWatchers = new HashMap<>();
     private Set<PinWatcher> waitingForPinWatchersForBackgroundUpdate;
 
     @Inject
-    public WatchManager(DatabaseManager databaseManager, ChanLoaderFactory chanLoaderFactory, WakeManager wakeManager, PageRequestManager pageRequestManager) {
+    public WatchManager(
+            DatabaseManager databaseManager,
+            ChanLoaderFactory chanLoaderFactory,
+            WakeManager wakeManager,
+            PageRequestManager pageRequestManager,
+            ThreadSaveManager threadSaveManager) {
         //retain local references to needed managers/factories/pins
         this.databaseManager = databaseManager;
         this.chanLoaderFactory = chanLoaderFactory;
         this.wakeManager = wakeManager;
         this.pageRequestManager = pageRequestManager;
+        this.threadSaveManager = threadSaveManager;
 
         databasePinManager = databaseManager.getDatabasePinManager();
+        databaseSavedThreadManager = databaseManager.getDatabaseSavedThreadmanager();
+
         pins = databaseManager.runTask(databasePinManager.getPins());
```

Comment:
TODO: After loading pins and savedThread check whether or not all saved threads are stopped or fully downloaded so we don't start a service just to stop it right away (because we already have a similar check in the service).